MonoGame.Framework/Audio/SoundEffectInstancePool.cs

```diff
@@ -140,13 +140,26 @@ internal static void Update()
                     Add(inst);
                     continue;
                 }
-                else if (inst._effect.IsDisposed)
+
+                x++;
+            }
+        }
+
+        /// <summary>
+        /// Iterates the list of playing instances, stop them and return them to the pool if they are instances of the given SoundEffect.
+        /// </summary>
+        /// <param name="effect">The SoundEffect</param>
+        internal static void StopPooledInstances(SoundEffect effect)
+        {
+            SoundEffectInstance inst = null;
+
+            for (var x = 0; x < _playingInstances.Count; x++)
```

Comment:
This won't iterate correctly.  The x++ here along with the x++ at the end of the loop statement will skip every second playing instance.  If an instance is found and removed via the Add(inst) call, the next two instances will be skipped.  Remove the x++ from the for statement and it will work correctly.